test_integration/shared.js

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+/**
+ * @file `shared` exists so that integration tests can share data among each other.
+ *
+ * One example is the addresses of contracts that were deployed on the test
+ * environment.
+ *
+ * Due to node's caching behavior when loading modules, it always returns the
+ * same object for repeated calls to `require()`.
+ *
+ * It is important that every `require` is written exactly `shared`,
+ * case-sensitive!
+ */
+
+/**
+ * Contains all relevant data on a chain.
+ *
+ * @property {Web3} web3 A web3 object to access the chain.
+ * @property {Object} contractAddresses All addresses in sting format, indexed
+ *     by the contract name, as written in the solidity source file.
+ * @property {Object} contracts Instances of TruffleContracts. Indexed by their
+ *     contract name, e.g. `EIP20Gateway` or `Anchor`. Exception: `BaseToken`
+ *     and `BrandedToken`.
+ * @property {string} deployerAddress The address of the account that deployed
+ *     the contracts in the deployment phase.
+ */
+class Chain {
+    constructor() {
+        this.web3 = {};
+        this.contractAddresses = {};
+        this.contracts = {};
```

Comment:
This should probably have accounts of that chain.

Reply:
You can get the accounts from `web3`

Reply:
If it's available with shared, it's very convenient for the integration tests.  Just as truffle provide us with accounts right for unit tests. Similar concept.

Reply:
Should I add more things from web3 (and maybe the contract artifacts or instances) to shared as well?

Reply:
We already have artifacts and instance(Per chain) in the shared, I believe. Am i missing something here? 🤔

Reply:
I mean details from within the artifacts or instances? For easier access from tests.

Reply:
No, I feel other details are test specific. Accounts are must have thing for all the tests.

Reply:
I realized I have added this in Stake and mint integration test.  We can skip for this PR 👍